pycloudlib/azure/cloud.py

```diff
@@ -665,15 +766,21 @@ def launch(
                 f"{self._type} launch requires image_id param."
                 f" Found: {image_id}"
             )
+        if not ip_addresses:
+            ip_addresses = list([None])
+        if not network_interface_clients:
+            network_interface_clients = list([None])
+
+        assert len(network_interface_clients) >= len(ip_addresses)
```

Comment:
Users can provide less IP addresses than network interfaces to be used, so that pycloudlib will create the missing ones. I think:

1. We should probably document this fact in the function.
2. Raise a proper exception, allowing users to catch it.